### ISSUE TITLE
Permission denied while hosts.present on win32

### ISSUE THREAD

**Issue by mpasternak** [saltstack/salt#18998]
Hi,

this is the traceback I get when I try to add hosts file entry with Windows 8.1 . Salt version 2014.7.0

```
thinkpad.win32:
----------
          ID: master
    Function: host.present
        Name: proxy.local
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "salt/state.py", line 1533, in call
                File "salt/states/host.py", line 78, in present
                File "salt/modules/hosts.py", line 221, in add_host
                File "salt/modules/hosts.py", line 235, in _write_hosts
                File "salt/utils/__init__.py", line 1065, in fopen
              IOError: [Errno 13] Permission denied: 'C:\\Windows\\System32\\drivers\\etc\\hosts'
     Started: 20:34:32.186000
    Duration: 16.0 ms
     Changes:   
----------
```

**Comment by ssgward** [comment]
@mpasternak - is this still an issue for you? or have you gotten past the original problem?

**Comment by mpasternak** [comment]
Well, nothing changed, it still does not work. Was there a new salt release
in the meantime?

2015-01-16 18:14 GMT+01:00 ssgward <notifications@github.com>:

> @mpasternak <https://github.com/mpasternak> - is this still an issue for
> you? or have you gotten past the original problem?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/saltstack/salt/issues/18998#issuecomment-70287098>.
>



-- 
Pozdrawiam,
Michał Pasternak
📱 +48 793 66 87 33

**Comment by ssgward** [comment]
We have version 2014.7.1 out. This may have been addressed for you via other bug fixes, but not certain since we can't duplicate, yet, in house. Meanwhile, will try other things to attempt to duplicate.

**Comment by mpasternak** [comment]
Hi,

I've installed 2014.7.1 on Windows 8.1 64-bit,

I have the same error, PermissionDenied while salt client tries to
overwrite hosts file.

... unless I turned off my antivirus software (Kaspersky Internet Security).

Seems like KIS treats hosts file in a special way, preventing its
modification by system service. After disabling KIS protection, salt-call
worked OKAY.

On the other hand, I can easily edit hosts file in a Notepad application
launched in admin mode.

Perhaps a service running in background is treated in a different way, than
a GUI application, when it comes to hosts file.

Anyway, this does not seem a Salt problem any longer. Definitely an
interesting case, maybe someone will want to investigate it in the future.

If you ask me, this issue can be closed. Thank you for your assistance and
congrats, you're working on a great product.



2015-01-30 23:04 GMT+01:00 ssgward <notifications@github.com>:

> We have version 2014.7.1 out. This may have been addressed for you via
> other bug fixes, but not certain since we can't duplicate, yet, in house.
> Meanwhile, will try other things to attempt to duplicate.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/saltstack/salt/issues/18998#issuecomment-72278277>.
>



-- 
Pozdrawiam,
Michał Pasternak
📱 +48 793 66 87 33
